### PR TITLE
kubeadm: kubeadm join test cmds for new flags

### PR DESCRIPTION
**Pull request by dmmcquay**
**What this PR does / why we need it**: Adding test-cmds for new kubeadm join flags. 

Adding tests is a WIP from #34136

**Special notes for your reviewer**: /cc @luxas 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/test/cmd/join_test.go [hunk 1]
```diff
@@ -46,3 +46,148 @@ func TestCmdJoinConfig(t *testing.T) {
...
+				actual,
+				rt.expected,
+				(actual == nil),
+			)
+		}
+		kubeadmReset()
+	}
+}
+
+func TestCmdJoinDiscoveryToken(t *testing.T) {
```
Comment (luxas) [inline cmd/kubeadm/test/cmd/join_test.go]: I think all the three funcs can be combined into a TestCmdJoinTokenArgs or similar

Comment (dmmcquay) [inline cmd/kubeadm/test/cmd/join_test.go]: The other test-cmds follow this similar format, test for each flag of a subcommand.

Comment (luxas) [inline cmd/kubeadm/test/cmd/join_test.go]: Yes, but I just thought that it sounds like a little too much duplication really, what do you think?
They all test the same type of tokens anyway (e.g. foo isn't a valid token)

Comment (errordeveloper) [inline cmd/kubeadm/test/cmd/join_test.go]: We should probably use [subtests](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks).

**Comment by dmmcquay** [comment]
being moved to https://github.com/kubernetes/kubernetes/pull/44628

**Pull request closed without merging**